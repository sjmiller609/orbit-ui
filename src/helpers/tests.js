import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
