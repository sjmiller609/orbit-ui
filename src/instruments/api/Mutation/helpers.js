const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
