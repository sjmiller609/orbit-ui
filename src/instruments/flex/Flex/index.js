import React from 'react'
import PropTypes from 'prop-types'

import classnames from 'classnames'
import s from './styles.scss'

const Flex = ({
  children,
  className,
  passRef,
  justify,
  align,
  wrap,
  auto,
  flow,
}) => {
  return (
    <div
      ref={passRef}
      className={classnames(
        s.flex,
        flow === 'row' ? s.row : s.box,
        s[justify],
        align ? s['align-' + align] : null,
        wrap ? s.wrap : null,
        auto ? s.auto : null,
        className
      )}>
      {Array.isArray(children) ? children.map(el => el) : children}
    </div>
  )
}

Flex.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.element,
    PropTypes.string,
  ]),
  className: PropTypes.string,
  passRef: PropTypes.func,
  justify: PropTypes.string,
  align: PropTypes.string,
  flow: PropTypes.string,
  wrap: PropTypes.bool,
  auto: PropTypes.bool,
}

export default Flex
