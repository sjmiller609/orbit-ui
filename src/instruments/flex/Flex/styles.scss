@import 'src/styles/mixins.scss';

@mixin flex($flow) {
  // width: inherit;
  // height: inherit;
  display: flex;
  flex-flow: $flow;
}

@mixin justify($justify: center) {
  justify-content: $justify;
}

@mixin align($align: center) {
  align-items: $align;
}

@mixin wrap($wrap: nowrap) {
  flex-wrap: $wrap;
}

.flex {
  @include wrap;
  @include justify;
  @include align;
}

.row {
  @include flex(row);
}

.box {
  @include flex(column);
}

// auto
.auto {
  flex: none;
  height: auto;
  width: auto;
}

.full {
  flex: auto;
  width: 100%;
  height: 100%;
}

// wrap
.wrap {
  @include wrap(wrap);
}

// justify content
.space-between {
  @include justify(space-between);
}
.space-around {
  @include justify(space-around);
}
.flex-end {
  @include justify(flex-end);
}
.flex-start {
  @include justify(flex-start);
}

// align content
.align-space-between {
  @include align(space-between);
}
.align-space-around {
  @include align(space-around);
}
.align-flex-end {
  @include align(flex-end);
}
.align-flex-start {
  @include align(flex-start);
}
