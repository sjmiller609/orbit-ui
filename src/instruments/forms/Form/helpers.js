// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
