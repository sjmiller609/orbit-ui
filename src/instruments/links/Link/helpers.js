// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
