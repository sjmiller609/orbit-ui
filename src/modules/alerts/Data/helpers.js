import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
