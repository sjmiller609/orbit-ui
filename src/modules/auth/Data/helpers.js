import React from 'react'

import { formErrors, Link } from 'instruments'

const errors = [
  {
    key: 'invalid password',
    name: 'password',
    error: 'Incorrect password',
  },
  {
    key: 'not found',
    name: 'email',
    error: "Hmm, we can't find that email",
  },
  {
    key: 'email already in use',
    name: 'email',
    error: 'That email is already taken.',
  },
  {
    key: 'no password credentials found',
    name: 'email',
    error: 'No password found. Did you mean to login with OAuth?',
  },
  {
    key: 'awaiting email confirmation',
    name: 'email',
    error: (
      <Link to="/resend">
        Your email is not yet verified. Please check your email for a
        verification link.<br /> [Click to resend]
      </Link>
    ),
  },
  {
    key: 'not associated with the specified invite',
    name: 'email',
    error: 'The email you entered does not match your invitation.',
  },
  {
    key: 'permissions',
    name: 'permissions',
    error: 'Contact your Workspace Admin to upgrade your permissions.',
  },
  {
    key: 'Email address in use',
    name: 'email',
    error: (
      <Link to="/login">
        A user with that email address already exists. Click here to login.
      </Link>
    ),
  },
]

export const handleError = error => formErrors({ error, errors })
