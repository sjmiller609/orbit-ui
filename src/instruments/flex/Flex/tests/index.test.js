import React from 'react'
import { mountWrap } from 'helpers/tests'
import Flex from '../index'

const props = {
  id: '1234',
  children: ['<p>This is a test</p>'],
  className: 'testClassString',
  passRef: jest.fn(),
  justify: 'space-between',
  align: 'center',
  flow: 'row',
  wrap: false,
  auto: false,
  full: false,
  onClick: jest.fn(),
}

describe('Flex', () => {
  it('renders correctly', () => {
    const wrapper = mountWrap(<Flex {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render full', () => {
    const wrapper = mountWrap(<Flex {...props} full={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('shoulnd render auto', () => {
    const wrapper = mountWrap(<Flex {...props} auto={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render wrap', () => {
    const wrapper = mountWrap(<Flex {...props} wrap={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('shoulnd render full, auto & wrap', () => {
    const wrapper = mountWrap(
      <Flex {...props} full={true} auto={true} wrap={true} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
