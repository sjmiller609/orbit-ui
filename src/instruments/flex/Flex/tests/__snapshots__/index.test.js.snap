// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex renders correctly 1`] = `
<Flex
  align="center"
  auto={false}
  className="testClassString"
  flow="row"
  full={false}
  id="1234"
  justify="space-between"
  onClick={[MockFunction]}
  passRef={
    [MockFunction] {
      "calls": Array [
        Array [
          <div
            class="flex row space-between align-center testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  wrap={false}
>
  <div
    className="flex row space-between align-center testClassString"
    id="1234"
    onClick={[MockFunction]}
  >
    &lt;p&gt;This is a test&lt;/p&gt;
  </div>
</Flex>
`;

exports[`Flex should render full 1`] = `
<Flex
  align="center"
  auto={false}
  className="testClassString"
  flow="row"
  full={true}
  id="1234"
  justify="space-between"
  onClick={[MockFunction]}
  passRef={
    [MockFunction] {
      "calls": Array [
        Array [
          <div
            class="flex row space-between align-center testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center full testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  wrap={false}
>
  <div
    className="flex row space-between align-center full testClassString"
    id="1234"
    onClick={[MockFunction]}
  >
    &lt;p&gt;This is a test&lt;/p&gt;
  </div>
</Flex>
`;

exports[`Flex should render wrap 1`] = `
<Flex
  align="center"
  auto={false}
  className="testClassString"
  flow="row"
  full={false}
  id="1234"
  justify="space-between"
  onClick={[MockFunction]}
  passRef={
    [MockFunction] {
      "calls": Array [
        Array [
          <div
            class="flex row space-between align-center testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center full testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center auto testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center wrap testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  wrap={true}
>
  <div
    className="flex row space-between align-center wrap testClassString"
    id="1234"
    onClick={[MockFunction]}
  >
    &lt;p&gt;This is a test&lt;/p&gt;
  </div>
</Flex>
`;

exports[`Flex shoulnd render auto 1`] = `
<Flex
  align="center"
  auto={true}
  className="testClassString"
  flow="row"
  full={false}
  id="1234"
  justify="space-between"
  onClick={[MockFunction]}
  passRef={
    [MockFunction] {
      "calls": Array [
        Array [
          <div
            class="flex row space-between align-center testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center full testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center auto testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  wrap={false}
>
  <div
    className="flex row space-between align-center auto testClassString"
    id="1234"
    onClick={[MockFunction]}
  >
    &lt;p&gt;This is a test&lt;/p&gt;
  </div>
</Flex>
`;

exports[`Flex shoulnd render full, auto & wrap 1`] = `
<Flex
  align="center"
  auto={true}
  className="testClassString"
  flow="row"
  full={true}
  id="1234"
  justify="space-between"
  onClick={[MockFunction]}
  passRef={
    [MockFunction] {
      "calls": Array [
        Array [
          <div
            class="flex row space-between align-center testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center full testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center auto testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center wrap testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
        Array [
          <div
            class="flex row space-between align-center wrap auto full testClassString"
            id="1234"
          >
            &lt;p&gt;This is a test&lt;/p&gt;
          </div>,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  wrap={true}
>
  <div
    className="flex row space-between align-center wrap auto full testClassString"
    id="1234"
    onClick={[MockFunction]}
  >
    &lt;p&gt;This is a test&lt;/p&gt;
  </div>
</Flex>
`;
